subt/artifacts.py

```diff
@@ -436,19 +436,39 @@ class ArtifactReporter(Node):
     def __init__(self, config, bus):
         super().__init__(config, bus)
         bus.register("artf_cmd")
+        self.repeat_report_sec = config.get('repeat_report_sec')
+        self.artf_xyz = []
+        self.artf_xyz_accumulated = []
 
-    def update(self):  # hack, this method should be called run instead!
-        channel = super().update()  # define self.time
-        assert channel == "artf_xyz", channel
-
+    def publish_artf(self, artf_xyz):
         print(self.time, "DETECTED:")
-        for artf_type, ix, iy, iz in self.artf_xyz:
+        for artf_type, ix, iy, iz in artf_xyz:
             print(" ", artf_type, ix/1000.0, iy/1000.0, iz/1000.0)
             s = '%s %.2f %.2f %.2f\n' % (artf_type, ix/1000.0, iy/1000.0, iz/1000.0)
             self.publish('artf_cmd', bytes('artf ' + s, encoding='ascii'))
-
         print('report completed')
 
+    def update(self):  # hack, this method should be called run instead!
+        channel = super().update()  # define self.time
+        assert channel in ["artf_xyz", "sim_time_sec"], channel
+
+        if channel == 'sim_time_sec':
+            if self.repeat_report_sec is None or self.sim_time_sec % self.repeat_report_sec != 0:
+                return channel
+            if len(self.artf_xyz) == 0:
+                return channel
+        elif channel == 'artf_xyz':
+            for item in self.artf_xyz:
```

Comment:
Here I was wondering, who on earth sets `self.artf_xyz`? Only later I realized it is set in `super().update()`. I know it was probably my design but today, I'd give myself :-1: for that for the lack of readability.

Reply:
:)